src/core/Akka/Actor/ActorCell.Children.cs

```diff
@@ -374,7 +374,7 @@ private IInternalActorRef MakeChild(Props props, string name, bool async, bool s
                     throw;
                 }
 
-                if (Mailbox != null)
+                if (Mailbox != null && IsFailed)
```

Comment:
The fix - we check to see if the parent is faulted before applying recursive suspend semantics to our children. Normally this code would normally only be hit if the parent was faulted and expected to restart (that was the only condition under which an actor could be suspended on the JVM.) But because of how we implement async / await, we should only trigger a suspension for a child if the suspension for the parent is because it's faulted - not because it's processing a `RunTask`.